core/commands/files/files.go

```diff
@@ -39,21 +41,33 @@ of consistency guarantees. If the daemon is unexpectedly killed before running
 'ipfs files flush' on the files in question, then data may be lost. This also
 applies to running 'ipfs repo gc' concurrently with '--flush=false'
 operations.
+
+The --cid-version and --hash-fun option only apply to newly created files
+and directories.  If not specified these proprieties are inhertied
+from the parent directory.
 `,
 	},
 	Options: []cmds.Option{
 		cmds.BoolOption("f", "flush", "Flush target and ancestors after write.").Default(true),
+		cmds.BoolOption("raw-leaves", "Use raw blocks for newly created leaf nodes. (experimental)"),
+		cmds.IntOption("cid-version", "cid-ver", "Cid version. Non-zero value will change default of 'raw-leaves' to true. (experimental)"),
+		cmds.StringOption("hash-fun", "Hash function to use. Will set Cid version to 1 if used. (experimental)"),
+		// ^^fixme: can't use just "hash" as the option name as the
+		// conflicts with "--hash" usage by the stat command, this is
+		// unfortunate as it creates an inconsistency with the "add"
+		// that uses "hash"
 	},
```

Comment:
@whyrusleeping here is the comment on why I can't use "hash" as the option name

Reply:
Can't we just have the `--hash` option for stat be different from the one on the other commands?

Reply:
We could, if you are willing to change the interface for stat.  That is something I didn't want to do.

Reply:
@whyrusleeping another option is to use `--mhash` instead of `--hash-fun` this is more specific (multi-hash) and is a good name for eventually making it a global option.

Reply:
I still don't think this is blocking anything. Stat is a separate command, it can have a flag that means something different, having those options on the `ipfs files` command doesnt make sense for a lot of cases too. `ipfs add`, which will at some point be moved under the `ipfs files` command also uses `--hash`. We need to stay consistent here.

Reply:
> Stat is a separate command, it can have a flag that means something different

Not with the current API.  Currently the `--hash-fun` function is a global option for all `ipfs files` command.  The `ipfs files stat` command` also has the `--hash` option.  It creates this error when the `stat` command is used:
```
$ ipfs files stat
Error: Option name 'hash' used multiple times
```
To resolve this conflict I need to either move the `--hash` option into each of the subcommands, or rename the `--stat` option in `ipfs files stat` to something else.  If I move the `--hash` option to each of the subcommands I should likely move the `--raw-leaves` and `--cid-ver` option also to be consistent.

Reply:
> Currently the --hash-fun function is a global option for all ipfs files command. 

dont do that.

> I need to either move the --hash option into each of the subcommands

Yes, do that.

Reply:
Done.